### PR TITLE
Update release workflows

### PR DESCRIPTION
**Pull request by Link-**
- resolves https://github.com/github/c2c-actions-runtime/issues/2221
- resolves https://github.com/github/c2c-actions-runtime/issues/2194
- resolves https://github.com/github/c2c-actions-runtime/issues/2212

### Summary

Following the transfer of `actions-runner-controller` to the `actions` organisation, the controller and runner release workflows have been broken.

This PR intends to fix those workflows while introducing the least amount of changes. The fact that packages (all images pushed to GHCR) remain under [the old organisation](https://github.com/orgs/actions-runner-controller/packages) – which is intentional to avoid breaking end-user environments – introduced unnecessary complexity to these workflows.

As we work on improving our posture, this PR remedies the situation and provides a mechanism to publish changes until the permanent fixes are introduced.

The behaviour of these workflows is fully described here: https://github.com/actions-runner-controller/releases

### Caveats

For all of these workflows, the image pushing after build has not been tested (to avoid polluting registries). This needs to be verified after we merge.

### TODO before merge

- [x] Commit all TODO suggestions
- [x] Test & confirm the [publish-chart.yaml workflow](https://github.com/actions/actions-runner-controller/blob/c407b33040afd6ca6ad6347105a30f8a94ac8824/.github/workflows/publish-chart.yaml) is working as expected
- [x] Remove all references to `set-output`: [ref1](https://github.com/actions/actions-runner-controller/blob/c407b33040afd6ca6ad6347105a30f8a94ac8824/.github/workflows/publish-chart.yaml#L102)
- [x] Update `DOCKER_ACCESS_TOKEN` and `DOCKER_USER` secrets in this repo and in actions-runner-controller/releases

### CODE REVIEW

File: .github/workflows/publish-chart.yaml [hunk 1]
```diff
@@ -107,8 +109,11 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       contents: write  # for helm/chart-releaser-action to push chart release and create a release
+    env:
+      CHART_TARGET_ORG: actions-runner-controller
+      CHART_TARGET_REPO: charts-test-repo
```
Comment (Link-) [inline .github/workflows/publish-chart.yaml]:
```suggestion
      CHART_TARGET_REPO: actions-runner-controller.github.io
```

**TODO:** this is needs to be committed before merging this PR.

File: .github/workflows/publish-canary.yaml [hunk 2]
```diff
@@ -19,10 +21,14 @@ on:
       - 'LICENSE'
       - 'Makefile'
 
+env:
+  PUSH_TO_REGISTRIES: false
```
Comment (Link-) [inline .github/workflows/publish-canary.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```
**TODO:** This needs to be committed before merging this PR.

File: .github/workflows/runners.yaml [hunk 3]
```diff
@@ -28,94 +30,31 @@ env:
   RUNNER_VERSION: 2.299.1
   DOCKER_VERSION: 20.10.21
   RUNNER_CONTAINER_HOOKS_VERSION: 0.1.3
-  DOCKERHUB_USERNAME: summerwind
+  TARGET_ORG: actions-runner-controller
+  PUSH_TO_REGISTRIES: false
```
Comment (Link-) [inline .github/workflows/runners.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```
**TODO:** This needs to be committed before merging this PR.

File: .github/workflows/publish-arc.yaml [hunk 4]
```diff
@@ -35,36 +48,34 @@ jobs:
...
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           make github-release
 
-      - name: Setup Docker Environment
-        uses: ./.github/actions/setup-docker-environment
+      - name: Get Token
+        id: get_workflow_token
+        uses: peter-murray/workflow-application-token-action@v1
```
Comment (Link-) [inline .github/workflows/publish-arc.yaml]: **TODO:** Replace `v1` with commit sha (version pinning)

File: .github/workflows/publish-chart.yaml [hunk 5]
```diff
@@ -120,8 +125,54 @@ jobs:
           git config user.name "$GITHUB_ACTOR"
           git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
 
-      - name: Run chart-releaser
+      - name: Get Token
+        id: get_workflow_token
+        uses: peter-murray/workflow-application-token-action@v1
```
Comment (Link-) [inline .github/workflows/publish-chart.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/publish-canary.yaml [hunk 6]
```diff
@@ -34,26 +40,19 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Setup Docker Environment
-        id: vars
-        uses: ./.github/actions/setup-docker-environment
+      - name: Get Token
+        id: get_workflow_token
+        uses: peter-murray/workflow-application-token-action@v1
```
Comment (Link-) [inline .github/workflows/publish-canary.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/runners.yaml [hunk 7]
```diff
@@ -28,94 +30,31 @@ env:
...
-
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-
-      - name: Setup Docker Environment
-        uses: ./.github/actions/setup-docker-environment
+      - name: Get Token
+        id: get_workflow_token
+        uses: peter-murray/workflow-application-token-action@v1
```
Comment (Link-) [inline .github/workflows/runners.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/publish-chart.yaml [hunk 8]
```diff
@@ -120,8 +125,54 @@ jobs:
...
+          path: ${{ env.CHART_TARGET_REPO }}
+          ref: ${{ env.CHART_TARGET_BRANCH }}
+          token: ${{ steps.get_workflow_token.outputs.token }}
+
+      - name: Copy index.yaml
+        run: |
+          cp ${{ github.workspace }}/index.yaml ${{ env.CHART_TARGET_REPO }}/index.yaml
+      
+      - name: Commit and Push
+        uses: stefanzweifel/git-auto-commit-action@v4
```
Comment (vanZeben) [inline .github/workflows/publish-chart.yaml]: Can we pin this version SHA as well please 🙏🏻

File: .github/workflows/publish-chart.yaml [hunk 9]
```diff
@@ -120,8 +125,54 @@ jobs:
...
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.CHART_TARGET_ORG }}/${{ env.CHART_TARGET_REPO }}
+          path: ${{ env.CHART_TARGET_REPO }}
+          ref: ${{ env.CHART_TARGET_BRANCH }}
+          token: ${{ steps.get_workflow_token.outputs.token }}
+
+      - name: Copy index.yaml
+        run: |
+          cp ${{ github.workspace }}/index.yaml ${{ env.CHART_TARGET_REPO }}/index.yaml
```
Comment (Link-) [inline .github/workflows/publish-chart.yaml]:
```suggestion
          cp ${{ github.workspace }}/index.yaml ${{ env.CHART_TARGET_REPO }}/actions-runner-controller/index.yaml
```
The index.yaml file should be inside the actions-runner-controller folder

File: .github/workflows/runners.yaml [hunk 10]
```diff
@@ -25,97 +27,36 @@ on:
       - '!**.md'
 
 env:
+  # Safeguard to prevent pushing images to registeries after build  
+  PUSH_TO_REGISTRIES: false
```
Comment (Link-) [inline .github/workflows/runners.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```

**TODO:** This needs to be committed before merging this PR.

**Review by mumoshu — DISMISSED** [review]
@Link- I'm continually impressed by your awesome work! LGTM.
Let me know if you need quick re-approval after you added the commits for the CHART_TARGET_REPO change before merging.

**Comment by Link-** [comment]
You're too kind @mumoshu I've committed the changes and would appreciate one final ✅

**Review by vanZeben — APPROVED** [review]

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks again for your efforts ☺️

**Pull request merged**